src/shogun/util/factory.h

```diff
@@ -164,6 +171,39 @@ namespace shogun
 				return std::make_shared<StringFeatures<char>>(
 				    file, alphabet_type);
 			}
+			case PT_UINT16:
+			{
+				return std::make_shared<StringFeatures<uint16_t>>(
+				    file, alphabet_type);
+			}
+			case PT_UINT32:
+			{
+				return std::make_shared<StringFeatures<uint32_t>>(
+				    file, alphabet_type);
+			}
+			case PT_UINT64:
+			{
+				return std::make_shared<StringFeatures<uint64_t>>(
+				    file, alphabet_type);
+			}
+			default:
+				not_implemented(SOURCE_LOCATION);
+			}
+
+			return nullptr;
```

Comment:
@karlnapf  not sure if it makes sense to add all these overloads?

Reply:
you mean having all the possible instantiations? I think we generally should only ever have stuff that is actually used anywhere?

Reply:
I mean it only makes sense to represent a string with unsigned integer types right?